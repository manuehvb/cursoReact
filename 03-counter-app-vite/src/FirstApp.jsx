import PropTypes from 'prop-types';

const FirstApp = ({ title, subTitle, name}) => {


    return (
        <>
            <h1 data-testid="test-title"> {title} </h1>
            {/* <code>{JSON.stringify(newMessage)}</code> */}
            <p>{subTitle}</p>
            <p>{subTitle}</p>
            <p>{ name }</p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}

FirstApp.defaultProps = {
    title: 'No hay titulo',
    subTitle: 'No hay subtitulo',
    name: 'AAA'
}

export default FirstApp;