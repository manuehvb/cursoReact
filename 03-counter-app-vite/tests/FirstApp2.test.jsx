import { render, screen } from "@testing-library/react";
import FirstApp from "../src/FirstApp";

describe('Pruebas en <FirstApp />', () => {

    const title = 'Titulo';
    const subTitle = 'Subtítulo';

    test('debe hacer match co el snapshot', () => {
        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar el mensaje "Título"', () => {
        render(<FirstApp title={title} />);
        /* screen.debug(); */
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('debe mostrar el título en un h1', () => {
        render(<FirstApp title={title} />);
        /* screen.debug(); */
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

    test('debe mostrar el subtítulo enviado por props', () => {

        render(<FirstApp title={title} subTitle={subTitle} />);
        /* screen.debug(); */
        expect(screen.getAllByText(subTitle).length).toBe(2);
    });
});