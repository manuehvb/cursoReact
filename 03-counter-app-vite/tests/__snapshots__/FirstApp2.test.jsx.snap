// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe hacer match co el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Titulo
     
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    AAA
  </p>
</div>
`;
